test/PathUtils.test.ts

```diff
@@ -431,4 +431,29 @@ describe('PathUtils ->', () => {
         assert.equal(PathUtils.getParentNodePath('/foobar'), '');
         assert.equal(PathUtils.getParentNodePath('/foobar/a/xyz'), '/foobar/a');
     });
+
+    describe('toAemPath', () => {
+        it('transform remote react app path to AEM path', () => {
+            const AEM_ORIGIN = window.location.origin;
+            const REMOTE_APP_PAGE = '/page';
+            const AEM_ROOT_PATH = 'testproj/lang';
+            const AEM_PAGE_PATH = `/content/${AEM_ROOT_PATH}/page.html`;
+
+            // Transform to match AEM path when running on AEM instance
+            let newPathPattern = new RegExp(PathUtils.toAEMPath(REMOTE_APP_PAGE, AEM_ORIGIN, AEM_ROOT_PATH));
+
+            expect(AEM_PAGE_PATH).toMatch(newPathPattern);
+
+            const REMOTE_ORIGIN = 'http://mydomain';
+            const windowSpy: jest.SpyInstance = jest.spyOn(global, 'window', 'get');
+
+            windowSpy.mockImplementation(() => ({ location: { origin: REMOTE_ORIGIN } }));
+
+            // return actual path when app is running on a remote domain
+            newPathPattern = new RegExp(PathUtils.toAEMPath(REMOTE_APP_PAGE, AEM_ORIGIN, AEM_ROOT_PATH));
+            windowSpy.mockRestore();
+
+            expect(REMOTE_APP_PAGE).toMatch(newPathPattern);
+        });
+    });
```

Comment:
we are missing tests:
- when `isLoadedInAEM` is false
- when `isLoadedInAEM` is true, and wcm mode is not `edit`